src/libraries/System.Text.Json/src/System/Text/Json/Nodes/JsonArray.cs

```diff
@@ -175,13 +175,7 @@ internal JsonArray(JsonElement element, JsonNodeOptions? options = null) : base(
         [RequiresDynamicCode(JsonValue.CreateDynamicCodeMessage)]
         public void Add<T>(T? value)
         {
-            JsonNode? nodeToAdd = value switch
-            {
-                null => null,
-                JsonNode node => node,
-                _ => JsonValue.Create(value, Options)
-            };
-
+            JsonNode? nodeToAdd = ConvertFromValue(value, Options);
             Add(nodeToAdd);
```

Comment:
> Add(nodeToAdd); [](http://example.com/codeflow?start=11&length=16)

If we have `nodeToAdd` is `null`. do we still need to call Add(...)?

Reply:
Yes. `JsonNode` uses `null` to represent `null` JSON values so it needs to be added in that case as well.